revised NG and OGS retirement schedule

ssattler opened the pull request:
Revised ReEDS generator database to reflect PA's NG and OGS retirement schedule for 35 year plant life as per: https://ucsusa-my.sharepoint.com/:x:/g/personal/parbaje_ucs_org/EbfbXamwBSxMsjoLcmRcYREB3VW0qcC7d_939CZevHDSfg?e=RKmWT3&xsdata=MDV8MDJ8U1NhdHRsZXJAdWNzLm9yZ3w5OGYyZTZkNjQ3YWM0ZjAwMDY2YzA4ZGRjNGE0YmM0ZnxiY2U0MTc1YjZjOTY0YjRkYWY3NTBmMWJjZDI0NjY3N3wwfDB8NjM4ODgyOTM0MDE1NDkxMTAwfFVua25vd258VFdGcGJHWnNiM2Q4ZXlKRmJYQjBlVTFoY0draU9uUnlkV1VzSWxZaU9pSXdMakF1TURBd01DSXNJbEFpT2lKWGFXNHpNaUlzSWtGT0lqb2lUV0ZwYkNJc0lsZFVJam95ZlE9PXwwfHx8&sdata=dmF6OWNSLzdXVnpPQWRQVXRrUGg3ZXVvckxnTTVzQytEaE8yRUNnak1mND0%3d

## Summary

## Technical Details
### Implementation notes

### Additional changes (if any)

### Switches added/removed/changed (if any)

### Issues resolved (if any)

### Known incompatibilities (if any)

### Relevant sources or documentation (if any)

### Is the model cleaner/better/faster/smaller than before? If so, how?

## Validation / Comparison Report
### Link to report and/or example plots

### (if pertinent) How big is the default run folder (runs/{base}_ref_seq/) before and after the change?

### (if pertinent) What is the run time for ref_seq on the same machine before and after the change?

ssattler committed:
revised NG and OGS retirement schedule

Revised ReEDS generator database to reflect PA's NG and OGS retirement schedule for 35 year plant life as per:
https://ucsusa-my.sharepoint.com/:x:/g/personal/parbaje_ucs_org/EbfbXamwBSxMsjoLcmRcYREB3VW0qcC7d_939CZevHDSfg?e=RKmWT3&xsdata=MDV8MDJ8U1NhdHRsZXJAdWNzLm9yZ3w5OGYyZTZkNjQ3YWM0ZjAwMDY2YzA4ZGRjNGE0YmM0ZnxiY2U0MTc1YjZjOTY0YjRkYWY3NTBmMWJjZDI0NjY3N3wwfDB8NjM4ODgyOTM0MDE1NDkxMTAwfFVua25vd258VFdGcGJHWnNiM2Q4ZXlKRmJYQjBlVTFoY0draU9uUnlkV1VzSWxZaU9pSXdMakF1TURBd01DSXNJbEFpT2lKWGFXNHpNaUlzSWtGT0lqb2lUV0ZwYkNJc0lsZFVJam95ZlE9PXwwfHx8&sdata=dmF6OWNSLzdXVnpPQWRQVXRrUGg3ZXVvckxnTTVzQytEaE8yRUNnak1mND0%3d

ssattler committed:
revised d_solve to limit gas

Revised the d_solveprep.gms file to disallow gas in LA starting in 2030 for entire projection.

samgdotson commented on d_solveprep.txt:
I think you accidentally created a new file rather than editing an existing file (I can’t see which lines are different). I believe it should be either a .gms or .gams file extension rather than .txt.

ssattler replied:
oops. I added the revised file just now.

samgdotson commented on inputs/capacity_exogenous/ReEDS_generator_database_final_EIA-NEMS.csv:
```diff
@@ -2662,7 +2662,7 @@ o-g-s,p53,136,99.7,2030,2030,2030,2030,2030,1963,TRUE,13888,p20009,Barton County
 gas-ct,p53,143,14,2043,2043,2043,2043,2043,1988,TRUE,19416,p20021,Cherokee County,KS,37.073,94.699,1322,1322,2,5860,SWPP ,1239,11,Riverton,1,CTX,ET,1,0,9,18,18,11,4,6,1,16.3,16,0,0,0,0,1988,11,12, ,0,0,0,0,0,0,  ,0,0.006,0.054,0.105,0.352,0.358,0.002,0,0.003,0.008,0.009,0.011,0.005,0,GI,DS,0,0.91,0.15,0,7.017535275,0.34360217,0,1,0.476,0,0,0,0,0,0,0,0,0,0.35,0,0,0,0.9,N ,N ,0,0,0,0,0,0,0,0,0,0,0, ...
 gas-cc,p53,143,149,2062,2062,2062,2062,2062,2007,TRUE,7328,p20021,Cherokee County,KS,37.073,94.699,1323,1323,1,5860,SWPP ,1239,12,Riverton,1,CCG,EC,1,0,5,18,18,11,4,6,1,148.8,180,844.3,0,4.412,10068.5,2007,5,12, ,0,0,0,0,0,0,  ,0,0.51,0.425,0.393,0.471,0.436,0.407,0.571,0.623,0.548,0.396,0.393,0.44,0.525,GF,0,0,1,0,0,0.75683865,8.987410604,0,1,0.08,0,0,93.3,0.46,0.05,0,0,0,0.16,0.35,0,0.54,0,0.9, ...
-gas-cc,p53,143,117,2071,2071,2071,2071,2071,2016,TRUE,7328,p20021,Cherokee County,KS,37.073,94.699,1323,1323,2,5860,SWPP ,1239,2-Dec,Riverton,1,CCG,EC,1,0,5,18,18,11,4,6,1,118.8,117,882.8,0,4.128,9220.9,2016,5,12, ,0,0,0,0,0,0,  ,0,0.51,0.51,0.472,0.565,0.523,0.407,0.571,0.623,0.548,0.396,0.393,0.44,0.525,GF,0,0,1,0,0,0.75683865,8.987410604,0,1,0.043,1,2,129.6,0.64,0.05,0,0,0,0.08,0.35,0,0.61,0,0 ...
+gas-cc,p53,143,117,2071,2071,2071,2071,2071,2016,TRUE,7328,p20021,Cherokee County,KS,37.073,94.699,1323,1323,2,5860,SWPP ,1239,45993,Riverton,1,CCG,EC,1,0,5,18,18,11,4,6,1,118.8,117,882.8,0,4.128,9220.9,2016,5,12, ,0,0,0,0,0,0,  ,0,0.51,0.51,0.472,0.565,0.523,0.407,0.571,0.623,0.548,0.396,0.393,0.44,0.525,GF,0,0,1,0,0,0.75683865,8.987410604,0,1,0.043,1,2,129.6,0.64,0.05,0,0,0,0.08,0.35,0,0.61,0,0 ...
```
What does “2-DEC” mean? And what does changing it to “45993” do? Are these date codes? Just want to make sure the intended column was modified.

ssattler replied:
That is an Excel error. The change should have never been made.

samgdotson replied:
which version is the correct version?

ssattler replied:
I don't see the 2_DEC in my version of this file. It should be 45993. I think it's the unit identifier within the plant.

ssattler replied:
Crap. I just looked at the latest and greatest NREL version of the database file. Somewhere along the line I converted this file to all "general" and it changed the date value of these cells to a numerical value. NREL's version has these cells as dates. While I don't think that this is correct (these are the unit identifiers of the parts of the plants listed), we might as well match NREL for the purpose of modeling. I'll change these cells back to dates.

samgdotson commented on inputs/capacity_exogenous/ReEDS_generator_database_final_EIA-NEMS.csv:
Given the number of generators that retire in 2049, maybe we should revisit reporting out to 2050?

samgdotson reviewed (changes requested):
The generator database looks OK — just need to confirm one thing. The d_solveprep.txt needs to be fixed before I can merge this.

ssattler committed:
revised d_solve to limit gas in LA

revised d_solveprep.gms to disallow new gas builds in LA from 2030-2050.

samgdotson reviewed (changes requested):
Thanks for replacing the gms file. It looks like this version doesn’t actually have the wind and storage capacity limits for LA. The values are actually “???” Can you replace those values as well?

ssattler committed:
revised d_solveprep limit LA gas

Previous d_solveprep file had wind and storage limits with ???. The LA gas limits need to be run off of the LA RPS scenario that does not include the wind and storage limits. This version removes the wind and storage limits.

ssattler commented:
I'm glad that this was caught, because the gas limits in LA needs to be run with the RPS version of the LA model. I removed the wind and storage capacity limits in the d_solveprep file.

samgdotson commented:
I'm pretty sure the RPS version of the model still includes limits on the growth rates of wind and storage. Let's confirm with @parbaje.

samgdotson commented:
Nevermind, I checked the `fy25-la-cap` file and it doesn't have the growth limits. That was for the `fy25-la-bau` scenario.

samgdotson committed:
Delete d_solveprep.txt

samgdotson commented on inputs/capacity_exogenous/ReEDS_generator_database_final_EIA-NEMS.csv:
```diff
@@ -3709,8 +3709,8 @@ o-g-s,p103,217,2,2013,2013,2013,2013,2013,1966,TRUE,12197,p26163,Wayne County,MI
 o-g-s,p103,217,2,2013,2013,2013,2013,2013,1966,TRUE,12197,p26163,Wayne County,MI,42.176,83.534,1808,1808,1,5109,MISO ,1727,3,Dayton (MI),1,CTO,ET,1,0,9,5,5,7,3,4,1,2,2,0,0,0,0,1966,4,11, ,0,0,0,0,0,0,  ,0,0,0,0,0,0,0,0,0,0,0,0,0,0,DS,0,0,1,0,0,5.481526534,1.230331313,0,1,2.505,0,0,0,0,0,0,0,0,0,0.35,0,0,0,0.9,N ,N ,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,5109,o-g-s_n_fsa,n,fsa,0,1,Existing Units,Oil  ...
-gas-ct,p103,217,63,2055,2055,2055,2055,2055,2000,TRUE,12947,p26163,Wayne County,MI,42.295,83.102,1809,1809,1,5109,MISO ,1728,1-Dec,Delray,1,CTG,ET,1,0,9,5,5,7,3,4,1,71.1,79,0,0,0,0,2000,2,12, ,0,0,0,0,0,0,  ,0,0.018,0.014,0.013,0.014,0.017,0.017,0.03,0.031,0.035,0.02,0.013,0.012,0.013,GI,0,0,1,0,0,0.947143212,0.34360217,0,1,0.04,0,0,300.5,1.48,0.05,0,0,0,0.13,0.35,0,0.88,0,0.9,N ,N ,0,0,0,0,0,0,0 ...
-gas-ct,p103,217,64,2055,2055,2055,2055,2055,2000,TRUE,12947,p26163,Wayne County,MI,42.295,83.102,1809,1809,1,5109,MISO ,1728,1-Nov,Delray,1,CTG,ET,1,0,9,5,5,7,3,4,1,71.1,80,0,0,0,0,2000,2,12, ,0,0,0,0,0,0,  ,0,0.018,0.014,0.013,0.014,0.017,0.017,0.03,0.031,0.035,0.02,0.013,0.012,0.013,GI,0,0,1,0,0,0.947143212,0.34360217,0,1,0.037,0,0,294.2,1.45,0.05,0,0,0,0.12,0.35,0,0.87,0,0.9,N ,N ,0,0,0,0,0,0, ...
+gas-ct,p103,217,63,2055,2055,2055,2055,2055,2000,TRUE,12947,p26163,Wayne County,MI,42.295,83.102,1809,1809,1,5109,MISO ,1728,45992,Delray,1,CTG,ET,1,0,9,5,5,7,3,4,1,71.1,79,0,0,0,0,2000,2,12, ,0,0,0,0,0,0,  ,0,0.018,0.014,0.013,0.014,0.017,0.017,0.03,0.031,0.035,0.02,0.013,0.012,0.013,GI,0,0,1,0,0,0.947143212,0.34360217,0,1,0.04,0,0,300.5,1.48,0.05,0,0,0,0.13,0.35,0,0.88,0,0.9,N ,N ,0,0,0,0,0,0,0 ...
+gas-ct,p103,217,64,2055,2055,2055,2055,2055,2000,TRUE,12947,p26163,Wayne County,MI,42.295,83.102,1809,1809,1,5109,MISO ,1728,45962,Delray,1,CTG,ET,1,0,9,5,5,7,3,4,1,71.1,80,0,0,0,0,2000,2,12, ,0,0,0,0,0,0,  ,0,0.018,0.014,0.013,0.014,0.017,0.017,0.03,0.031,0.035,0.02,0.013,0.012,0.013,GI,0,0,1,0,0,0.947143212,0.34360217,0,1,0.037,0,0,294.2,1.45,0.05,0,0,0,0.12,0.35,0,0.87,0,0.9,N ,N ,0,0,0,0,0,0, ...
```
@ssattler, these rows were also changed from a date to a raw number... is the numerical version correct?

ssattler replied:
The numerical version is correct.

samgdotson commented on inputs/capacity_exogenous/ReEDS_generator_database_final_EIA-NEMS.csv:
```diff
@@ -4141,7 +4141,7 @@ gas-cc,p43,127,117,2031,2031,2031,2031,2031,1954,TRUE,7450,p27037,Dakota County,
 coalolduns,p43,127,153,2015,2015,2015,2015,2015,1960,TRUE,11140,p27037,Dakota County,MN,44.811,93.25,1976,1976,1,13781,MISO ,1904,4,Black Dog,1,CSU,H1,1,0,1,3,3,10,4,5,1,163.2,153,1442.9,6.784,2.806,16416.6,1960,9,4, ,0,0,1,98,0,332.3,  ,0,0.5,0,0,0,0,0,0,0,0,0,0,0,0,H1,DS,WD,0.99,0.01,0,1.007193918,21.67395557,8.735507539,1,0.297,0,0,90,0.44,0.05,0.724,21.4,0.43,0.73,0.35,0,0.54,0,0.9,N ,N ,0,0, ...
 gas-cc,p43,127,165,2031,2031,2031,2031,2031,2002,TRUE,7450,p27037,Dakota County,MN,44.811,93.25,1977,1977,1,13781,MISO ,1904,5,Black Dog,1,CCG,EC,1,0,5,3,3,10,4,5,1,187.9,181,935.7,0,3.42,8995.2,2002,6,12, ,0,0,0,0,0,0,  ,0,0.453,0.358,0.367,0.319,0.489,0.556,0.612,0.714,0.653,0.403,0.418,0.23,0.33,GF,0,0,1,0,0,5.733058547,4.342080426,0,1,0.1,0,2,81.2,0.4,0.05,0,0,0,0.2,0.35,0,0.52,0,0.9,N ,N ,0, ...
-gas-ct,p43,127,212,2058,2058,2058,2058,2058,2018,TRUE,10745,p27037,Dakota County,MN,44.811,93.25,1978,1978,1,13781,MISO ,1904,1-Jun,Black Dog,1,CTG,ET,1,0,9,3,3,10,4,5,1,238,228,0,0,0,0,2018,4,12, ,0,0,0,0,0,0,  ,0,0.161,0.172,0.159,0.14,0.141,0.136,0.18,0.228,0.203,0.144,0.137,0.131,0.156,GI,0,0,1,0,0,1.74861846,1.121119057,0,1,0.309,0,0,57.8,0.28,0.05,0,0,0,0.56,0.35,0,0.47,0,0.9,N ,N ,0,0,0,0, ...
+gas-ct,p43,127,212,2058,2058,2058,2058,2058,2018,TRUE,10745,p27037,Dakota County,MN,44.811,93.25,1978,1978,1,13781,MISO ,1904,45809,Black Dog,1,CTG,ET,1,0,9,3,3,10,4,5,1,238,228,0,0,0,0,2018,4,12, ,0,0,0,0,0,0,  ,0,0.161,0.172,0.159,0.14,0.141,0.136,0.18,0.228,0.203,0.144,0.137,0.131,0.156,GI,0,0,1,0,0,1.74861846,1.121119057,0,1,0.309,0,0,57.8,0.28,0.05,0,0,0,0.56,0.35,0,0.47,0,0.9,N ,N ,0,0,0,0, ...
```
Here, as well.

samgdotson commented on inputs/capacity_exogenous/ReEDS_generator_database_final_EIA-NEMS.csv:
```diff
@@ -9879,12 +9879,12 @@ gas-ct,p126,332,86.9,2046,2046,2046,2046,2046,1991,TRUE,13656,p34011,Cumberland
-gas-cc,p55,147,70,2052,2052,2052,2052,2052,1997,TRUE,7372,p29097,Jasper County,MO,37.066,94.614,4835,4835,2,22500,SWPP ,7296,2-Feb,State Line Combined,1,CCG,EC,1,0,5,18,18,11,4,6,0.4,72,80,657,0,2.81,8700.8,1997,6,12, ,0,0,0,0,0,0,  ,0,0.551,0.591,0.63,0.479,0.363,0.414,0.691,0.723,0.689,0.625,0.428,0.4,0.571,GF,0,0,1,0,0,0.577629499,8.987410604,0,1,0.015,0,2,75,0.37,0.05,0,0,0,0.03,0.35,0,0.51,0 ...
-gas-cc,p55,147,111.6,2056,2056,2056,2056,2056,2001,TRUE,7372,p29097,Jasper County,MO,37.066,94.614,4835,4835,3,5860,SWPP ,7296,3-Feb,State Line Combined,1,CCG,EC,1,0,5,18,18,11,4,6,0.6,123.9,111.6,657,0,2.81,8700.8,2001,6,12, ,0,0,0,0,0,0,  ,0,0.551,0.591,0.63,0.479,0.363,0.414,0.691,0.723,0.689,0.625,0.428,0.4,0.571,GF,0,0,1,0,0,0.577629499,8.987410604,0,1,0.043,0,2,69,0.34,0.05,0,0,0,0.09,0.35, ...
-gas-cc,p55,147,74.4,2056,2056,2056,2056,2056,2001,TRUE,7372,p29097,Jasper County,MO,37.066,94.614,4835,4835,3,22500,SWPP ,7296,3-Feb,State Line Combined,1,CCG,EC,1,0,5,18,18,11,4,6,0.4,82.6,74.4,657,0,2.81,8700.8,2001,6,12, ,0,0,0,0,0,0,  ,0,0.551,0.591,0.63,0.479,0.363,0.414,0.691,0.723,0.689,0.625,0.428,0.4,0.571,GF,0,0,1,0,0,0.577629499,8.987410604,0,1,0.043,0,2,69,0.34,0.05,0,0,0,0.09,0.35,0, ...
+gas-cc,p55,147,70,2056,2056,2056,2056,2056,2001,TRUE,7372,p29097,Jasper County,MO,37.066,94.614,4835,4835,1,22500,SWPP ,7296,45689,State Line Combined,1,CCG,EC,1,0,5,18,18,11,4,6,0.4,72.4,80,657,0,2.81,8700.8,2001,6,12, ,0,0,0,0,0,0,  ,0,0.551,0.591,0.63,0.479,0.363,0.414,0.691,0.723,0.689,0.625,0.428,0.4,0.571,GF,0,0,1,0,0,0.577629499,8.987410604,0,1,0.015,0,2,75,0.37,0.05,0,0,0,0.03,0.35,0,0.51 ...
+gas-cc,p55,147,105,2056,2056,2056,2056,2056,2001,TRUE,7372,p29097,Jasper County,MO,37.066,94.614,4835,4835,1,5860,SWPP ,7296,45689,State Line Combined,1,CCG,EC,1,0,5,18,18,11,4,6,0.6,108.6,120,657,0,2.81,8700.8,2001,6,12, ,0,0,0,0,0,0,  ,0,0.551,0.591,0.63,0.479,0.363,0.414,0.691,0.723,0.689,0.625,0.428,0.4,0.571,GF,0,0,1,0,0,0.577629499,8.987410604,0,1,0.015,0,2,75,0.37,0.05,0,0,0,0.03,0.35,0,0. ...
+gas-cc,p55,147,105,2052,2052,2052,2052,2052,1997,TRUE,7372,p29097,Jasper County,MO,37.066,94.614,4835,4835,2,5860,SWPP ,7296,45690,State Line Combined,1,CCG,EC,1,0,5,18,18,11,4,6,0.6,108,120,657,0,2.81,8700.8,1997,6,12, ,0,0,0,0,0,0,  ,0,0.551,0.591,0.63,0.479,0.363,0.414,0.691,0.723,0.689,0.625,0.428,0.4,0.571,GF,0,0,1,0,0,0.577629499,8.987410604,0,1,0.015,0,2,75,0.37,0.05,0,0,0,0.03,0.35,0,0.51 ...
+gas-cc,p55,147,70,2052,2052,2052,2052,2052,1997,TRUE,7372,p29097,Jasper County,MO,37.066,94.614,4835,4835,2,22500,SWPP ,7296,45690,State Line Combined,1,CCG,EC,1,0,5,18,18,11,4,6,0.4,72,80,657,0,2.81,8700.8,1997,6,12, ,0,0,0,0,0,0,  ,0,0.551,0.591,0.63,0.479,0.363,0.414,0.691,0.723,0.689,0.625,0.428,0.4,0.571,GF,0,0,1,0,0,0.577629499,8.987410604,0,1,0.015,0,2,75,0.37,0.05,0,0,0,0.03,0.35,0,0.51,0 ...
+gas-cc,p55,147,111.6,2056,2056,2056,2056,2056,2001,TRUE,7372,p29097,Jasper County,MO,37.066,94.614,4835,4835,3,5860,SWPP ,7296,45691,State Line Combined,1,CCG,EC,1,0,5,18,18,11,4,6,0.6,123.9,111.6,657,0,2.81,8700.8,2001,6,12, ,0,0,0,0,0,0,  ,0,0.551,0.591,0.63,0.479,0.363,0.414,0.691,0.723,0.689,0.625,0.428,0.4,0.571,GF,0,0,1,0,0,0.577629499,8.987410604,0,1,0.043,0,2,69,0.34,0.05,0,0,0,0.09,0.35, ...
```
Also these rows.

samgdotson commented on inputs/capacity_exogenous/ReEDS_generator_database_final_EIA-NEMS.csv:
```diff
@@ -11756,10 +11756,10 @@ hydEND,p127,340,1,2024,2024,2024,2024,2024,1924,TRUE,,p36091,Saratoga County,NY,
-lfill-gas,p103,215,0.8,2091,2091,2091,2091,2091,1991,TRUE,13500,p26037,Clinton County,MI,42.791,84.696,5566,5566,1,55858,MISO ,10532,3-Feb,Grand River,1,MSW,MS,3,0,4,5,5,7,3,4,1,0.8,0.8,0,0,0,0,1991,4,12, ,0,0,0,0,0,0,  ,0,0.395,0.448,0.453,0.443,0.444,0.459,0.424,0.417,0.416,0.397,0.412,0.414,0.435,SW,0,0,1,0,0,3.416,202.444,0,1,0.043,0,0,0,0,0,0,0,0,0,0.35,0,0,0,0.9,N ,N ,0,0,0,0,0,0,0,0,0,0,0, ...
-lfill-gas,p103,215,0.8,2096,2096,2096,2096,2096,1996,TRUE,13500,p26037,Clinton County,MI,42.791,84.696,5566,5566,2,55858,MISO ,10532,4-Feb,Grand River,1,MSW,MS,3,0,4,5,5,7,3,4,1,0.8,0.8,0,0,0,0,1996,1,12, ,0,0,0,0,0,0,  ,0,0.395,0.448,0.453,0.443,0.444,0.459,0.424,0.417,0.416,0.397,0.412,0.414,0.435,SW,0,0,1,0,0,3.518,206.632,0,1,0.043,0,0,0,0,0,0,0,0,0,0.35,0,0,0,0.9,N ,N ,0,0,0,0,0,0,0,0,0,0,0, ...
-lfill-gas,p103,215,0.8,2097,2097,2097,2097,2097,1997,TRUE,13500,p26037,Clinton County,MI,42.791,84.696,5566,5566,3,55858,MISO ,10532,5-Feb,Grand River,1,MSW,MS,3,0,4,5,5,7,3,4,1,0.8,0.8,0,0,0,0,1997,8,12, ,0,0,0,0,0,0,  ,0,0.395,0.448,0.453,0.443,0.444,0.459,0.424,0.417,0.416,0.397,0.412,0.414,0.435,SW,0,0,1,0,0,3.535,208.32,0,1,0.043,0,0,0,0,0,0,0,0,0,0.35,0,0,0,0.9,N ,N ,0,0,0,0,0,0,0,0,0,0,0,0 ...
+lfill-gas,p103,215,0.8,2091,2091,2091,2091,2091,1991,TRUE,13500,p26037,Clinton County,MI,42.791,84.696,5566,5566,1,55858,MISO ,10532,45689,Grand River,1,MSW,MS,3,0,4,5,5,7,3,4,1,0.8,0.8,0,0,0,0,1991,4,12, ,0,0,0,0,0,0,  ,0,0.395,0.448,0.453,0.443,0.444,0.459,0.424,0.417,0.416,0.397,0.412,0.414,0.435,SW,0,0,1,0,0,3.423,202.558,0,1,0.043,0,0,0,0,0,0,0,0,0,0.35,0,0,0,0.9,N ,N ,0,0,0,0,0,0,0,0,0,0,0, ...
+lfill-gas,p103,215,0.8,2091,2091,2091,2091,2091,1991,TRUE,13500,p26037,Clinton County,MI,42.791,84.696,5566,5566,1,55858,MISO ,10532,45691,Grand River,1,MSW,MS,3,0,4,5,5,7,3,4,1,0.8,0.8,0,0,0,0,1991,4,12, ,0,0,0,0,0,0,  ,0,0.395,0.448,0.453,0.443,0.444,0.459,0.424,0.417,0.416,0.397,0.412,0.414,0.435,SW,0,0,1,0,0,3.416,202.444,0,1,0.043,0,0,0,0,0,0,0,0,0,0.35,0,0,0,0.9,N ,N ,0,0,0,0,0,0,0,0,0,0,0, ...
+lfill-gas,p103,215,0.8,2096,2096,2096,2096,2096,1996,TRUE,13500,p26037,Clinton County,MI,42.791,84.696,5566,5566,2,55858,MISO ,10532,45692,Grand River,1,MSW,MS,3,0,4,5,5,7,3,4,1,0.8,0.8,0,0,0,0,1996,1,12, ,0,0,0,0,0,0,  ,0,0.395,0.448,0.453,0.443,0.444,0.459,0.424,0.417,0.416,0.397,0.412,0.414,0.435,SW,0,0,1,0,0,3.518,206.632,0,1,0.043,0,0,0,0,0,0,0,0,0,0.35,0,0,0,0.9,N ,N ,0,0,0,0,0,0,0,0,0,0,0, ...
+lfill-gas,p103,215,0.8,2097,2097,2097,2097,2097,1997,TRUE,13500,p26037,Clinton County,MI,42.791,84.696,5566,5566,3,55858,MISO ,10532,45693,Grand River,1,MSW,MS,3,0,4,5,5,7,3,4,1,0.8,0.8,0,0,0,0,1997,8,12, ,0,0,0,0,0,0,  ,0,0.395,0.448,0.453,0.443,0.444,0.459,0.424,0.417,0.416,0.397,0.412,0.414,0.435,SW,0,0,1,0,0,3.535,208.32,0,1,0.043,0,0,0,0,0,0,0,0,0,0.35,0,0,0,0.9,N ,N ,0,0,0,0,0,0,0,0,0,0,0,0 ...
```
And here

samgdotson reviewed (changes requested):
So, it looks like there are two kinds of changes happening.

1. Operating/retirement dates -- I assume. These changes are expected.
2. Some date field (e.g., 1-Mar) being converted to a numerical field. Please confirm that the numerical field is correct and that the "format" in excel is all "general."

Thanks!

ssattler committed:
changed database gas units back to dates

Changed the revised ReEDS_generator_database file unit identifiers back to dates to match NREL's database.

ssattler committed:
Revised format of lfg units

Further revised the ReEDS_generator_database file lfg unit identifiers back to dates, to match NREL database.

ssattler commented:
Is there anything else outstanding for this run? I'm getting a little lost in the conversation above.

samgdotson commented on d_solveprep.gms:
```diff
@@ -102,11 +102,15 @@ trans_cost_cap_fin_mult_noITC(t) = round(trans_cost_cap_fin_mult_noITC(t),3) ;
 INV.fx(i,v,r,t_notallowed)$[wind(i)$ r_st(r,'LA')] = 0 ;
+* Restrict GAS in LA, per PA - SSS 7/14/25
+Set t_gasnotallowed(t) / 2030*2050 / ;
+m_required_prescriptions("gas",r,t_gasnotallowed)$r_st(r,"LA") = 0 ;
+INV.fx(i,v,r,t_gasnotallowed)$[gas(i)$ r_st(r,'LA')] = 0 ;
```
I’m not sure “gas” or “gas(i)” will capture all of the technologies we wish to capture (e.g., o-g-s). 

Would you mind checking the [subset of techs](https://github.com/ucsusa/ReEDS-2.0/blob/0b57ae128bf9295dee0cf0218fe5f5fc800d26d9/b_inputs.gms#L276) and make sure we don’t need another one? It’s also possible that we could be more refined in our exclusions… (and allow CCS).

samgdotson replied:
```suggestion
INV.fx(i,v,r,t_gasnotallowed)$[gas(i)$ r_st(r,'LA')] = 0 ;
*Restrict O-G-S in LA - SGD 7/21/25
INV.fx(i,v,r,t_gasnotallowed)$[ogs(i)$ r_st(r,'LA')] = 0 ;
```

I think setting the `m_required_prescription` to zero is redundant for model years that aren't "historical." I'm also suggesting we add another line to restrict OGS in LA as well.

samgdotson committed:
Update d_solveprep.gms

samgdotson reviewed (approved):
Looks good, reviewed together.
